syntax = "proto3";
import "google/protobuf/any.proto";

option go_package = "./kre";

enum MessageType {
  OK = 0;
  ERROR = 1;
  EARLY_REPLY = 2;
  EARLY_EXIT = 3;
}

message KreNatsMessage { 
  string request_id = 1;
  google.protobuf.Any payload = 2;
  string error = 3;
  string from_node = 4;
  MessageType message_type = 5;
}
